File: test_3_4/workers.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.concurrent.TimeUnit;

class SimpleThread extends Thread {
    private int id;

    public SimpleThread(int id) {
        this.id = id;
    }

    public void run() {
        try {
            for (int i = 0; i < 5; i++) {
                executeTask(i);
                Thread.sleep(2000);
            }
        } catch (Exception e) {
        }
    }

    private void executeTask(int idx) {
        for (int j = 0; j < 10; j++) {
            try {
                Connection conn = DriverManager.getConnection("jdbc:h2:mem:testdb");
                PreparedStatement ps = conn.prepareStatement("SELECT * FROM sample WHERE id = ?");
                ps.setInt(1, j);
                ps.executeQuery();
                ps.close();
                conn.close();
            } catch (Exception e) {
            }
        }
    }
}

class Job implements Runnable {
    private int id;

    public Job(int id) {
        this.id = id;
    }

    public void run() {
        try {
            for (int i = 0; i < 5; i++) {
                queryDB(i);
                Thread.sleep(1500);
            }
        } catch (Exception e) {
        }
    }

    private void queryDB(int val) {
        try {
            Connection conn = DriverManager.getConnection("jdbc:h2:mem:testdb");
            PreparedStatement ps = conn.prepareStatement("SELECT name FROM sample WHERE key_col = ?");
            ps.setInt(1, val);
            ps.executeQuery();
            ps.close();
            conn.close();
        } catch (Exception e) {
        }
    }
}

public class MainClass {
    public static void main(String[] args) throws InterruptedException {
        for (int i = 0; i < 5; i++) {
            new SimpleThread(i).start();
        }

        ExecutorService pool = Executors.newFixedThreadPool(3);

        for (int i = 0; i < 5; i++) {
            pool.submit(new Job(i));
        }

        pool.shutdown();
        pool.awaitTermination(10, TimeUnit.MINUTES);
    }
}

File: test_3_4/workers.py
import threading
import sqlite3
import time
from concurrent.futures import ThreadPoolExecutor

class TaskThread(threading.Thread):
    def __init__(self, identifier):
        super().__init__()
        self.identifier = identifier

    def run(self):
        for i in range(5):
            self.db_query(i)
            time.sleep(2)

    def db_query(self, idx):
        for _ in range(10):
            conn = sqlite3.connect(":memory:")
            cursor = conn.cursor()
            cursor.execute("CREATE TABLE IF NOT EXISTS sample (id INTEGER PRIMARY KEY, value TEXT)")
            cursor.execute("INSERT INTO sample (value) VALUES ('test')")
            cursor.execute("SELECT * FROM sample WHERE id = ?", (idx,))
            cursor.fetchall()
            cursor.close()
            conn.close()

class Worker:
    def __init__(self, identifier):
        self.identifier = identifier

    def execute(self):
        for i in range(5):
            self.make_query(i)
            time.sleep(1.5)

    def make_query(self, val):
        conn = sqlite3.connect(":memory:")
        cursor = conn.cursor()
        cursor.execute("CREATE TABLE IF NOT EXISTS sample (id INTEGER PRIMARY KEY, value TEXT)")
        cursor.execute("INSERT INTO sample (value) VALUES ('worker')")
        cursor.execute("SELECT value FROM sample WHERE id = ?", (val,))
        cursor.fetchall()
        cursor.close()
        conn.close()

def main():
    threads = [TaskThread(i) for i in range(5)]
    for t in threads:
        t.start()

    with ThreadPoolExecutor(max_workers=3) as executor:
        workers = [Worker(i) for i in range(5)]
        for w in workers:
            executor.submit(w.execute)

    for t in threads:
        t.join()

if __name__ == "__main__":
    main()
